tlm_teamd/teamdctl_mgr.cpp

```diff
@@ -181,7 +197,32 @@ TeamdCtlDump TeamdCtlMgr::get_dump(const std::string & lag_name)
         }
         else
         {
-            SWSS_LOG_ERROR("Can't get dump for LAG '%s'. Skipping", lag_name.c_str());
+            // In case of failure and retry flag is set, check if it fails for MAX_RETRY times.
```

Comment:
in line 196, if there is no issue, do we need to clear out no_of_retry to 0 there?

Reply:
I had the static variables here and this approach to clear here originally.

On further testing, I moved it out to remove API , as I found that the get_dump for that particular lag_name was not getting called again as the remove lag request ( which comes in between ) clears off that entry for the lag_name from the m_handlers. So the no_of_retry doesn't get cleared.

This particular race condition don't happen in the add_lag sequence as the way we create the teamd process + entry in APP DB is a good. This issue happens with **remove lag** case -- as we remove teamd process instance first .. and the APP DB update comes very delayed.

Reply:
can you publish new changes so that i can take a look at?

Reply:
I have added the check to clear the no_of_retry here, in the good case for the lag in get_dump. Thanks